packages/dataviews/src/normalize-fields.ts

```diff
@@ -1,15 +1,17 @@
 /**
  * Internal dependencies
  */
-import type { Field, NormalizedField } from './types';
+import type { Field, Item, NormalizedField } from './types';
 
 /**
  * Apply default values and normalize the fields config.
  *
  * @param fields Fields config.
  * @return Normalized fields config.
  */
-export function normalizeFields( fields: Field[] ): NormalizedField[] {
+export function normalizeFields< T extends Item >(
+	fields: Field< T >[]
+): NormalizedField< T >[] {
```

Comment:
I'm starting to enjoy typescript in some cases.

But this kind of changes is what makes it unbearable to a lot of people. It just makes the code way less readable. I do understand the reasoning though and don't have solutions though.

Reply:
I can't argue with you, it's not pretty. But it does facilitate deeper type checking of the items, which I have found helpful.